sale_automatic_workflow_with_delay/models/automatic_workflow_job.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import models
+
+
+class AutomaticWorkflowJob(models.Model):
+    _inherit = "automatic.workflow.job"
+
+    def run_with_workflow(self, sale_workflow):
+        if sale_workflow.validate_order_with_delay:
+            self.env = self.env(
+                context=dict(self.env.context, sale_workflow_validate_with_delay=True)
+            )
+        return super(AutomaticWorkflowJob, self).run_with_workflow(sale_workflow)
+
+    def _validate_sale_orders(self, order_filter):
+        if self.env.context.get("sale_workflow_validate_with_delay"):
```

Comment:
Do we really need all these overrides?
Can't we have a specific method to call (eg: run_with_workflow_with_delay() which runs only the allowed ones? 
Only validate orders in this case.